src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsWriteStream.cs

```diff
@@ -0,0 +1,274 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+
+namespace Microsoft.Data.SqlClientX.IO
+{
+    /// <summary>
+    /// A stream abstraction over the TDS protocol which works to write the TDS packet
+    /// This stream will take care of wrapping the bytes with the TDS header. 
+    /// </summary>
+    internal class TdsWriteStream : Stream, ITdsWriteStream
```

Comment:
We wont need the Read/Write functionality without the other one. 
The idea was to encapsulate the Read and Write of TDS stream separate, because they are sort of orthogonal, and I want to err on the side of separating these concerns in two different classes.

I could potentially merge these two in TdsStream itself.

I am not taking a hard stance on this topic. But I will leave it out of scope of this PR. 

I want to get my TdsReadStream PR in as well, and then when both exist, I will be happy to revisit and refactor, if we think that partial classes or putting both in TdsStream makes sense.